File: node/7/GreetMe/test_greet_me.js
const assert = require("assert").strict;
var func = require("./greet_me");

let ptr = [
    // Each test pattern
    {"JACK": "Hello Jack!"},
    {"ricky": "Hello Ricky!"}
];

ptr.forEach(function(p) {
    var inp = Object.keys(p)[0];
    var exp = p[inp];

    assert.equal(func(inp), exp);
})

File: python/7/GreetMe/test_greet_me.py
from unittest import TestCase
from unittest import main

from .greet_me import greet


class TestGreetMe(TestCase):
    def test_greet(self):
        ptr = [
            ('riley', 'Hello Riley!'),
            ('molly', 'Hello Molly!'),
            ('BILLY', 'Hello Billy!'),
        ]
        for inp, exp in ptr:
            with self.subTest(inp=inp, exp=exp):
                self.assertEqual(greet(name=inp), exp)


if __name__ == "__main__":
    main(verbosity=2)
